src/frontend/src/optimizer/plan_node/logical_join.rs

```diff
@@ -1232,6 +1232,15 @@ impl ToBatch for LogicalJoin {
 
 impl ToStream for LogicalJoin {
     fn to_stream(&self, ctx: &mut ToStreamContext) -> Result<PlanRef> {
+        if self
+            .on()
+            .conjunctions
+            .iter()
+            .any(|cond| cond.count_nows() > 0)
+        {
+            return Err(ErrorCode::NotSupported("now() expression in ON clause".to_string(), "please refer to https://www.risingwave.dev/docs/current/sql-pattern-temporal-filters/ for more information".to_string()).into());
```

Comment:
Can you explain more about why it's invalid? like https://github.com/risingwavelabs/risingwave/blob/103652136437d41812fb9482c70ad429f18a0c4c/src/frontend/src/optimizer/plan_node/logical_filter.rs#L204-L218